python/cuML/test/test_knn.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+from cuml import KNN as cuKNN
+from sklearn.neighbors import KDTree as skKNN
+import cudf
+import pandas as pd
+import numpy as np
+
+
+@pytest.mark.parametrize('input_type', ['dataframe', 'ndarray'])
+def test_knn_search(input_type):
```

Comment:
do you know if/how knn tests against results against sklearn? So far I've had the assumption that we don't need to do as much testing like dbscan/kmean new tests, but it'd be good to be certain of that.

Reply:
I did quick tests for this particular pull request because, from our conversation on Friday, the more comprehensive testing wasn't slated for 0.6. By the completion of issue #78, and for all algorithms/wrappers going forward, I'd really like to have algorithms that really stress the variability in different sets of data against sklearn's results.

Reply:
Then you'll like an upcoming PR for 0.6 that is going to increase pytest for **all** algorithms soon